pkg/setting/setting.go

```diff
@@ -450,15 +450,21 @@ type CommandLineArgs struct {
 func (cfg Cfg) parseAppUrlAndSubUrl(section *ini.Section) (string, string, error) {
 	appUrl := valueAsString(section, "root_url", "http://localhost:3000/")
 
-	if appUrl[len(appUrl)-1] != '/' {
-		appUrl += "/"
```

Comment:
Why do you change this behavior?

Reply:
> If the appUrl is "" you get a segfault instead of an error. Tracking that down is frustrating compared to a message that points to the configuration.

Very good point.

> appUrl[len(appUrl)-1] gets you the last byte of the string. That might be the string "/" (\x2F), but it could also be a lot of other values. The grafana HTTP server correctly serves whether there is a trailing slash and the end of the URL, subPath or not.
>
> That said, Google at least indexes "example.com/path" and "example.com/path/" as separate URLs, so I'm happy to change that back but include a size check on the config.

Yeah, the challenge here is whether internal services uses the appUrl value expecting there to be a trailing slash, that might require a lap through the codebase to make sure we don't do that (I believe you recently fixed a bunch of the internal use of appUrl that might've been cause for concern, so it might not be the case that there are any such places left, but if we reopen this or open another iteration of this, I'd like [myself or someone else] to make a bit of a thorough review for that with some sort of written record of what we looked at 🤔)